pkg/kubelet/cm/dra/plugin/plugins_store.go

```diff
@@ -43,26 +42,31 @@ func (s *pluginsStore) get(pluginName string) *plugin {
 
 // Set lets you save a DRA Plugin to the list and give it a specific name.
 // This method is protected by a mutex.
-func (s *pluginsStore) add(pluginName string, p *plugin) {
+func (s *pluginsStore) add(pluginName string, p *Plugin) (replaced bool) {
```

Comment:
I'm not a big fan of the `bool` return value. I'd prefer an explicit get check followed by an add that returns nothing. Same for the `delete()` call that returns itself or nil -- I'd rather it return nothing, forcing an explicit `get()` check before hand.

Reply:
Calling `get` followed by `add` is not atomic. Removing these return values leads to a potential race condition.

Reply:
Ack. I still don't like it, but it's better than forcing one to lock outside of the `add()` / `delete()` calls when there is nothing else that needs to be locked.